packages/graphiql/src/components/GraphiQL.tsx

```diff
@@ -1052,11 +1227,18 @@ export class GraphiQL extends React.Component<GraphiQLProps, GraphiQLState> {
       shouldPersistHeaders: Boolean(this.props.shouldPersistHeaders),
       documentAST: this.state.documentAST,
     };
-    if (this.state.headers && this.state.headers.trim().length > 2) {
-      fetcherOpts.headers = JSON.parse(this.state.headers);
-      // if state is not present, but props are
-    } else if (this.props.headers) {
-      fetcherOpts.headers = JSON.parse(this.props.headers);
+    try {
+      if (this.state.headers && this.state.headers.trim().length > 2) {
+        fetcherOpts.headers = JSON.parse(this.state.headers);
+        // if state is not present, but props are
+      } else if (this.props.headers) {
+        fetcherOpts.headers = JSON.parse(this.props.headers);
+      }
+    } catch (err) {
+      this.setState({
+        response: 'Introspection failed as headers are invalid.',
+      });
+      return;
```

Comment:
Note: we should have a refetch schema button

Reply:
I 100% agree, and a keyboard shortcut. We have a keyboard shortcut and a refresh button in the monaco editor demo already haha